tox.ini

```diff
@@ -25,98 +55,75 @@ deps =
     testing.mysqld
     webtest
-    # Python2 only deps
-    py{27}: enum34 mock
+    wrapt
+
+    ; Python 2.7 only deps
+    py{27}: enum34
+    py{27}: mock
 
     # pymysql deps
     py{27,34,35}: pymysql < 1.0.0
     py{36,37,38,39}: pymysql >= 1.0.0
 
-    # Python3.4 only deps
+    # Python 3.4 only deps
     py34: typing >= 3.7.4.3
 
-    # Python3.5+ only deps
+    # Python 3.5+ only deps
     py{35,36,37,38,39}: aiohttp >= 3.0.0
     py{35,36,37,38,39}: pytest-aiohttp
     py{35,36,37,38,39}: aiobotocore >= 0.10.0
 
-commands =
-    coverage erase
-    py{27,34}-default: coverage run --append --source aws_xray_sdk -m py.test tests --ignore tests/ext/aiohttp --ignore tests/ext/aiobotocore --ignore tests/ext/django --ignore tests/test_async_local_storage.py --ignore tests/test_async_recorder.py --ignore tests/ext/sqlalchemy_core
-    py{35,36,37,38,39}-default: coverage run --append --source aws_xray_sdk -m py.test tests --ignore tests/ext/django --ignore tests/ext/sqlalchemy_core
-    py{27,36,37,38,39}-default: coverage run --append --source aws_xray_sdk -m py.test tests/ext/sqlalchemy_core
-    py{34,35}-default: coverage run --append --source aws_xray_sdk -m py.test tests/ext/sqlalchemy_core/ --ignore tests/ext/sqlalchemy_core/test_sqlalchemy_core_2.py
-    django{22,30,31,32}: coverage run --append --source aws_xray_sdk -m py.test tests/ext/django
-    codecov
-
 setenv =
     DJANGO_SETTINGS_MODULE = tests.ext.django.app.settings
     AWS_SECRET_ACCESS_KEY = fake_key
     AWS_ACCESS_KEY_ID=fake_id
 
-[testenv:py35-aiohttp2]
-deps =
-    pytest > 3.0.0
-    aiohttp >= 2.3.0,<3.0.0
-    pytest-aiohttp
-    botocore
-    coverage==4.5.4
-
 commands =
-    py{35}: coverage run --source aws_xray_sdk -m py.test tests/ext/aiohttp --ignore tests/ext/aiohttp/test_client.py
+    coverage erase
 
-[testenv:py36-aiohttp2]
-deps =
-    pytest > 3.0.0
-    aiohttp >= 2.3.0,<3.0.0
-    pytest-aiohttp
-    botocore
-    coverage==4.5.4
+    ; Async methods are only available for python 3.5+
+    py{27,34}-core: coverage run --append --source aws_xray_sdk -m pytest --ignore tests/ext --ignore tests/test_async_local_storage.py --ignore tests/test_async_recorder.py
+    py{35,36,37,38,39}-core: coverage run --append --source aws_xray_sdk -m pytest --ignore tests/ext
 
-commands =
-    py{36}: coverage run --source aws_xray_sdk -m py.test tests/ext/aiohttp --ignore tests/ext/aiohttp/test_client.py
+    ext-aiobotocore: coverage run --append --source aws_xray_sdk -m pytest tests/ext/aiobotocore
 
-[testenv:py37-aiohttp2]
-deps =
-    pytest > 3.0.0
-    aiohttp >= 2.3.0,<3.0.0
-    pytest-aiohttp
-    botocore
-    coverage==4.5.4
+    ext-aiohttp: coverage run --append --source aws_xray_sdk -m pytest tests/ext/aiohttp
 
-commands =
-    py{37}: coverage run --source aws_xray_sdk -m py.test tests/ext/aiohttp --ignore tests/ext/aiohttp/test_client.py
+    ext-botocore: coverage run --append --source aws_xray_sdk -m pytest tests/ext/botocore
```

Comment:
This is just a passing thought so feel free to leave it as is!

I was looking at [the `coverage` docs](https://coverage.readthedocs.io/en/6.2/source.html#execution) for the `--source` flag, and I found this:

> the coverage run command will by default measure all code

> You can specify source to measure with the --source command-line switch,

I would've thought this meant by default we can omit `--source` if we are in the directory with the tests, but I am less confident after having [found this](https://coverage.readthedocs.io/en/6.2/#quick-start):

> To limit coverage measurement to code in the current directory, and also find files that weren’t executed at all, add the --source=. argument to your coverage command line.

So I was wondering if we would benefit from [following OpenTelemetry's example](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/dd72d94615a66105a5275efac08f1fd5c89c54b2/tox.ini#L244) and using the tox `changedir =` keyword to simplify this line:

```
changedir =
  ext-botocore: tests/ext/botocore
commands =
  ext-botocore: coverage run --append --source . -m pytest
```

It would be nice to test this without the `--source .` to see if that works? But otherwise I think splitting out into the `chagedir` so this line doesn't get to long and it is simply just `pytest` in the end.

(We can almost reduce _all_ the `ext-` tests to just run the same command if we do this, but I think we're blocked by the `py{34,35}-ext-sqlalchemy_core` tests which need the `--ignore` flag... maybe there is a way to make `tox factor` not match this... but I'm not aware of how 🙂 )

Reply:
Yeah, I'm not sure if I really understood the `--source` flag. I did try the `changedir` keyword but for some reason, pytest was unable to find any test cases. I experimented a little but then gave up. I agree that despite being some blocking test environments, using `changedir` would yield a cleaner tox.ini file.

Reply:
It's already incredibly clean with your changes! Thanks for looking into it :)